pkg/plugins/plugin.go

```diff
@@ -370,16 +367,17 @@ func (p *Plugin) subOutboxFunc(sub chan<- Event, funcs ...filters.FilterFunc) fu
 		// we are using a queue subscription to only deliver the work to one of the plugins,
 		// because they subscribe to a group by the plugin name.
 		s, err := sc.QueueSubscribe(p.runtime.Outbox, p.runtime.Name, func(m *stan.Msg) {
-			// this is recreating the messsage from the inbox
-			msg, err := message.FromByte(m.Data)
+			e := cloudevents.Event{}
+
+			err := json.Unmarshal(m.Data, &e)
 			if err != nil {
-				sub <- &MessageError{Code: ErrParse, Msg: err.Error()}
 				return
 			}
 
 			botMessage := new(pb.Message)
-			if err := p.marshaler.Unmarshal(msg, botMessage); err != nil {
+			if err := p.marshaler.Unmarshal(e, botMessage); err != nil {
```

Comment:
what do you think about abstracting away the cloudevents to just the marshaller?

i.e. `func (JSONMarshaler) Unmarshal([]bytes, v interface{}) (err error) `, and internally in there we can unmarshal into a cloudevent Event and then unmarshal the data into the specified event??